Adjust lombok annotations to reduce build warnings

ghubstan opened the pull request:
This change fixes the underlying problems behind five compiler warnings:  `Generating equals/hashCode implementation but without a call to superclass`.

Some core classes use `@Value` and `@Data` annotations where not necessary, for example, where `equals`, `hashCode`, and `toString` methods, and `final` class and field modifiers are defined by a developer.

From lombok classes and documents for the shortcut annotations [`@Value`](https://projectlombok.org/features/Value)  and [`@Data`]( https://projectlombok.org/features/Data), we find some implicit annotations applied by these shortcuts are not needed.  Adjustments are described below.

- `Tx` Replace `@Value` with `@Getter`.

- `TxOutput`  Replace `@Data` with `@Getter` and `@Setter`.

- `ChangeParamProposal`  Replace `@Value` with `@Getter`.

- `GetInventoryRequest`, `GetInventoryResponse`  An `@EqualsAndHashCode(callSuper = false) `is added to supress build warnings.

ghubstan committed:
Adjust lombok annotations to reduce build warnings

This change fixes the underlying problems behind five compiler warnings:
"Generating equals/hashCode implementation but without a call to superclass".

Some core classes use @Value and @Data annotations where not necessary, for
example, where equals, hashCode, and toString are implemented by the developer,
and where final class and field modifiers are defined by the developer.

From lombok classes and documents for the shortcut annotations
	@Value https://projectlombok.org/features/Value and
	@Data  https://projectlombok.org/features/Data,
we find some implicit annotations applied by these shortcuts are not needed.
Adjustments are described below.

- Tx Replace @Value with @Getter.

- TxOutput  Replace @Data with @Getter and @Setter.

- ChangeParamProposal  Replace @Value with @Getter.

- GetInventoryRequest, GetInventoryResponse
  An @EqualsAndHashCode(callSuper = false) is added to supress build
  warnings.

ghubstan committed:
Remove unused imports

sqrrm reviewed (commented):
Makes me nervous changing this magic annotations in the DAO module. It compiles nicely and seems to work fine.

This is the typical case where @ManfredKarrer has to review to make sure this is not breaking some hashes we rely on in the DAO.

chimp1984 commented on core/src/main/java/bisq/core/dao/state/model/blockchain/TxOutput.java:
```diff
@@ -36,7 +37,8 @@
 @Immutable
-@Data
+@Getter
+@Setter
```
There is no write access and its Immutable. The @Data was wrong and should have been a @Value annotation.
Can you remove the @Setter?

ghubstan replied:
Setter removed in commit c2c1ac2087a8862dbf02007adf299aba460f92f5.

chimp1984 commented on core/src/main/java/bisq/core/network/p2p/inventory/messages/GetInventoryRequest.java:
```diff
@@ -21,9 +21,11 @@
+import lombok.EqualsAndHashCode;
 import lombok.Value;
 
 @Value
```
That should also be replaced by a @Getter and @ToString.

ghubstan replied:
Commit  c2c1ac2 
Removes `@Value`, `@EqualsAndHashCode(callSuper = false)`
Adds `@Getter`,`@ToString`

I'm not absolutely certain this is what you asked for (by removing `@Value`), but keeping `@Value` , `@Getter`, `@ToString` is redundant because `@Getter`, `@ToString` are implicitly defined by `@Value`.

chimp1984 replied:
I meant that @Value produces Lombok warnings as well so replace it with EqualsAndHashCode, Getter and ToString

chimp1984 commented on core/src/main/java/bisq/core/network/p2p/inventory/messages/GetInventoryResponse.java:
```diff
@@ -28,9 +28,11 @@
+import lombok.EqualsAndHashCode;
 import lombok.Value;
 
 @Value
```
That should also be replaced by a @Getter and @ToString.

ghubstan replied:
Same as above...  commit  c2c1ac2 
Removes `@Value`, `@EqualsAndHashCode(callSuper = false)`
Adds `@Getter`,`@ToString`

I'm not absolutely certain this is what you asked for (by removing `@Value`), but keeping `@Value` , `@Getter`, `@ToString` is redundant because `@Getter`, `@ToString` are implicitly defined by `@Value`.

chimp1984 replied:
Ah sorry I see now the confusion as the marked line was EqualsAndHashCode instead of Value. My fault... sorry...

ghubstan committed:
Merge branch 'master' into fix-annotations

ghubstan committed:
Adjust class level annoations

as per suggested changes
    https://github.com/bisq-network/bisq/pull/5021#discussion_r550903787
    https://github.com/bisq-network/bisq/pull/5021#discussion_r550903843
    https://github.com/bisq-network/bisq/pull/5021#discussion_r550903860

chimp1984 commented on core/src/main/java/bisq/core/network/p2p/inventory/messages/GetInventoryRequest.java:
```diff
@@ -21,11 +21,11 @@
+import lombok.ToString;
 
-@Value
-@EqualsAndHashCode(callSuper = false)
```
EqualsAndHashCode should be left

ghubstan replied:
commit 0c9c96165b2df15d9bd4c8a671d3083192f11403

chimp1984 commented on core/src/main/java/bisq/core/network/p2p/inventory/messages/GetInventoryResponse.java:
```diff
@@ -28,11 +28,11 @@
+import lombok.ToString;
 
-@Value
-@EqualsAndHashCode(callSuper = false)
```
EqualsAndHashCode should be left

ghubstan replied:
commit 0c9c96165b2df15d9bd4c8a671d3083192f11403

ghubstan committed:
Re-add @EqualsAndHashCode

ManfredKarrer reviewed (approved):
utACK
